internal/serverinstall/nomad.go

```diff
@@ -661,14 +715,63 @@ func waypointNomadJob(c nomadConfig, rawRunFlags []string) *api.Job {
 			},
 		},
 	}
-	// Preserve disk, otherwise upgrades will destroy previous allocation and the
-	// disk along with it
-	tg.EphemeralDisk = &api.EphemeralDisk{
-		Sticky:  &[]bool{true}[0],
-		Migrate: &[]bool{true}[0],
+
+	// Preserve disk, otherwise upgrades will destroy previous allocation and the disk along with it
+	volumeRequest := api.VolumeRequest{
+		Type:     c.volumeType,
+		ReadOnly: false,
 	}
+
+	if strings.ToLower(c.volumeType) == "csi" {
+		volumeRequest.Source = "waypoint"
+		volumeRequest.AccessMode = "single-node-writer"
+		volumeRequest.AttachmentMode = "file-system"
+	} else {
```

Comment:
We should be explicit here and check for `c.VolumeType == "host"`. Then the else could be an error saying we got an unrecognized or unsupported volume type. Alternatively we could also do this near the beginning when we're checking for missing flags so we don't go through the whole process and error here 🤔

Reply:
Yeah the checks to make sure the flags match up are up https://github.com/hashicorp/waypoint/blob/d42221d58f1c481f139731d077caa1e9c31f0036/internal/serverinstall/nomad.go#L142

Reply:
Ah yeah, I mean we should error if they give us an unsupported volume type, like if they accidentally type `-nomad-volume-type = "something-not-real"`

Reply:
Hm, now that I've tried this locally, I wonder if it's clearer to eliminate the `-nomad-volume-type` all together? So if a user requests `-nomad-csi-volume-provider=<string>` or ` -nomad-host-volume=<string>` the type is implied by the flag requested? What do you think? 🤔

Reply:
I agree with @briancain, I'd set the type based on what other options they pass in (note also if you do that, you need to error out if they pass in both csi-volume-provider and a host-volume flags).